sdk/nodejs/image.ts

```diff
@@ -150,12 +150,15 @@ export class Image extends pulumi.CustomResource {
      */
     public /*out*/ readonly registryServer!: pulumi.Output<string>;
     /**
+     * **For pushed images:**
      * The manifest digest of an image pushed to a registry, of the format repository@<algorithm>:<hash>, e.g. `username/demo-image@sha256:a6ae6dd8d39c5bb02320e41abf00cd4cb35905fec540e37d306c878be8d38bd3`.
      * This reference is unique per image build and push. 
      * Only available for images pushed to a registry.
      * Use when passing a reference to a pushed image to container management resources.
+     *
+     * **Local-only images**For local images, this field is the image ID of the built local image, of the format <algorithm>:<hash>, e.g `sha256:826a130323165bb0ccb0374ae774f885c067a951b51a6ee133577f4e5dbc4119` 
      */
-    public /*out*/ readonly repoDigest!: pulumi.Output<string | undefined>;
+    public /*out*/ readonly repoDigest!: pulumi.Output<string>;
```

Comment:
💥 

There are no examples here that had to be (or were able to be!) changed due to this.  I wonder if we should add a test that actually uses `repoDigest` that shows that we've actually simplified interacting with this.  

Related, and somewhat orthogonal to this.  I assume it is not a breaking change to make an optional output required - we are still always returning data that was previously valid, there is just one of the values were used to return that we never do anymore.  But I do wonder if in practice this might be breaking in some of our stricter languages?  Like is the change in Go a breaking change for code that had to process this as a StringPtrOutput?  

I think the answer is that it is.  I don't think that should prevent us from making this change, but we should call out in CHANGELOG the change that might be required by code using this property.